apps/src/componentLibrary/chips/_Chip.tsx

```diff
@@ -0,0 +1,71 @@
+import React, {useEffect, useRef, memo} from 'react';
+
+const commonI18n = require('@cdo/locale');
+import moduleStyles from './chip.module.scss';
+
+interface ChipProps {
+  /** Chip label */
+  label: string;
+  /** Chip name*/
+  name: string;
+  /** Chip value */
+  value: string;
+  /** Chip checked state */
+  checked: boolean;
+  /** Chip required state */
+  required: boolean;
+  /** Chip disabled state */
+  disabled?: boolean;
+  /** Chip onChange handler*/
+  onCheckedChange: (checked: boolean) => void;
+}
+
+const Chip: React.FunctionComponent<ChipProps> = ({
+  label,
+  name,
+  value,
+  checked,
+  required,
+  disabled,
+  onCheckedChange,
+}) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    // Reset validity on every render so it gets checked again.
+    // Otherwise, removing the `required` attribute doesn't work as expected.
+    const input = inputRef.current;
+    if (input) {
+      input.setCustomValidity('');
+    }
+  }, [inputRef, onCheckedChange]);
+
+  return (
+    <div>
+      <label className={moduleStyles.chip}>
+        <input
+          ref={inputRef}
+          type="checkbox"
+          name={name}
+          value={value}
+          checked={checked}
+          required={required}
+          disabled={disabled}
+          onChange={e => {
+            onCheckedChange(e.target.checked);
+            // Reset validity so it gets checked again.
+            e.target.setCustomValidity('');
+          }}
+          onInvalid={e => {
```

Comment:
How does this work exactly? would it be better for the consumer to decide what message to display in an invalid state?

Reply:
I thought of it, but since that's the only use cale of invalid message for now and it don't looks like there'll be any other different messages I decided it would be easier to keep this string in the component so that we won't need to constantly pass the same prop (something like: ```invalidMessage={coomonI18n.chooseAtLeastOne()}```)

Reply:
Gotcha - that makes sense, though it also might be good to not take a dependency on localized strings so it's easier to pull these components out into a separate package if/when we want to. I'll leave it up to you!

Reply:
Yeah, I thought of it as well. Decided that once we'll be migrating to separate package - we'll either create a common local i18n strings(e.g. for form fields validation) for package or refactor component to use external strings